artemis-server/src/main/java/org/apache/activemq/artemis/core/server/management/impl/ManagementServiceImpl.java

```diff
@@ -390,6 +393,11 @@ public ICoreMessage handleMessage(Message message) throws Exception {
       reply.setType(Message.TEXT_TYPE);
       reply.setReplyTo(message.getReplyTo());
 
+      String correlationId = getCorrelationIdentity(message);
+      if (correlationId != null) {
+         MessageUtil.setJMSCorrelationID(reply, correlationId);
```

Comment:
Can we avoid using JMS in core. You should set correlation using core api

Reply:
e.g. what would occur for other protocols that in future we may want to support management service, that may not match JMS. 

It better to introduce correlationId get and set at message api level, e.g. in AMQP it would set in properties, for CORE could make it map to a _AMQ property. etc.

Reply:
@michaelandrepearce thanks for the feedback.   I'm inexperience in the Artemis code base but  think I understand what you mean.  I'll rework the patch once @clebertsuconic has had chance to comment on merits or otherwise of the use-case.

Reply:
@k-wall I never doubt the usecase.. I just wanted to have it documented for other users when they look on the JIRA / feature request.

one thing I wonder if this warrants documentation changes or not.